lib/components.js

```diff
@@ -0,0 +1,422 @@
+/* global window, document, $, HTMLElement, customElements, AudioContext, URL */
+const createSwarm = require('killa-beez')
+const getUserMedia = require('getusermedia')
+const xhr = require('xhr')
+const bel = require('bel')
+const mediaRecorder = require('media-recorder-stream')
+const FileWriteStream = require('filestream/write')
+
+const UserStorage = require('./storage')
+const views = require('./views')
+const connectAudio = require('./connectAudio')
+
+const byId = id => document.getElementById(id)
+const values = obj => Object.keys(obj).map(k => obj[k])
+const selector = exp => document.querySelector(exp)
+const selectall = exp => document.querySelectorAll(exp)
+
+const context = new AudioContext()
+const waudio = require('waudio')(context)
+
+function getRtcConfig (cb) {
+  xhr({
+    url: 'https://instant.io/rtcConfig',
+    timeout: 10000
+  }, (err, res) => {
+    if (err || res.statusCode !== 200) {
+      cb(new Error('Could not get WebRTC config from server. Using default (without TURN).'))
+    } else {
+      var rtcConfig
+      try {
+        rtcConfig = JSON.parse(res.body)
+      } catch (err) {
+        return cb(new Error('Got invalid WebRTC config from server: ' + res.body))
+      }
+      cb(null, rtcConfig)
+    }
+  })
+}
+
+const defaultSignalHost = 'https://signalexchange.now.sh'
+const defaultRoomHost = 'https://roomexchange.now.sh'
+
+class RollCall extends HTMLElement {
+  constructor (self) {
+    self = super(self)
+
+    console.log('asdfasdfasdf')
+
+    this.storage = new UserStorage()
+    this.masterSoundOutput = waudio(true)
+
+    this.mainContainer = bel`
+    <div id="main-container" class="ui main text container">
+      <div id="messages-container"></div>
+      <div id="tracks-container" class="ui items"></div>
+      <div id="audio-container" class="ui special cards"></div>
+    </div>`
+    this.appendChild(this.mainContainer)
```

Comment:
Adding children inside of the constructor of a custom element isn't valid.

You'll get an error like this: http://stackoverflow.com/questions/40181683/failed-to-execute-createelement-on-document-the-result-must-not-have-childr

If created via `document.createElement('roll-call')`.